packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -0,0 +1,444 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:ui';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_push_notifications/amplify_push_notifications.dart';
+import 'package:amplify_push_notifications/src/native_push_notifications_plugin.g.dart';
+import 'package:amplify_push_notifications/src/push_notifications_flutter_api.dart';
+import 'package:amplify_secure_storage/amplify_secure_storage.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'amplify_push_noitfications_impl_test.mocks.dart';
+import 'test_data/fake_amplify_configuration.dart';
+import 'test_data/fake_defective_configuration.dart';
+import 'test_data/fake_notification_messges.dart';
+import 'util.dart';
+
+void testGlobalCallbackFunction(PushNotificationMessage pushMessage) {}
+
+@GenerateMocks(
+  [
+    PushNotificationsHostApi,
+    ServiceProviderClient,
+    AmplifySecureStorage,
+    PluginUtilities,
+    AmplifyPushNotificationsFlutterApi,
+  ],
+)
+void main() {
+  const tokenReceivedEventChannel = EventChannel(
+    'com.amazonaws.amplify/push_notification/event/TOKEN_RECEIVED',
+  );
+
+  final log = <MethodCall>[];
+  late TestWidgetsFlutterBinding testWidgetsFlutterBinding;
```

Comment:
Should only be called once per file.

```suggestion
  final testWidgetsFlutterBinding = TestWidgetsFlutterBinding.ensureInitialized();
```